src/rabbit_queue_index.erl

```diff
@@ -490,12 +493,9 @@ start(DurableQueueNames) ->
 
 stop() -> rabbit_recovery_terms:stop().
 
-all_queue_directory_names(Dir) ->
-    case rabbit_file:list_dir(Dir) of
-        {ok, Entries}   -> [E || E <- Entries,
-                                 rabbit_file:is_dir(filename:join(Dir, E))];
-        {error, enoent} -> []
-    end.
+all_queue_directory_names() ->
+    QueuesBaseDir = queues_base_dir(),
+    filelib:wildcard(filename:join([QueuesBaseDir, "*", "queues", "*"])).
```

Comment:
See below. It would be less risky if we used `{node_database_dir}/vhosts/{vhost}/queues/{queue}` for path naming.